File: jobportal-frontend/jobportal-frontend/src/services/jobService.js
import api from "./api";

export const getAllJobs = async () => {
    const res = await api.get("/jobs");
    return res.data;
};

export const getJobById = async (id) => {
    const res = await api.get(`/jobs/${id}`);
    return res.data;
};

export const createJob = async (job) => {
    const res = await api.post("/jobs", job);
    return res.data;
};

export const updateJob = async (id, job) => {
    const res = await api.put(`/jobs/${id}`, job);
    return res.data;
};

export const closeJob = async (id) => {
    const res = await api.post(`/jobs/${id}/close`);
    return res.data;
};

File: jobportal/src/main/java/com/kiwisoft/jobportal/service/JobService.java
package com.kiwisoft.jobportal.service;

import com.kiwisoft.jobportal.dto.request.AssignPartnerRequest;
import com.kiwisoft.jobportal.dto.request.JobRequest;
import com.kiwisoft.jobportal.dto.request.UpdateJobRequest;
import com.kiwisoft.jobportal.dto.response.JobResponse;

import java.util.List;

public interface JobService {

    JobResponse createJob(JobRequest request);

    List<JobResponse> getAllJobs();

    JobResponse getJobById(Long id);

    JobResponse updateJob(
            Long id,
            UpdateJobRequest request
    );

    JobResponse closeJob(Long id);

    String assignPartner(
            Long jobId,
            AssignPartnerRequest request
    );

    List<JobResponse> getJobsForPartner(
            Long partnerId
    );

}